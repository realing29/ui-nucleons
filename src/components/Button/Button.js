import React from 'react';
import PropTypes from 'prop-types';

/**
 * Компонент кнопки
 */
export default class Button extends React.Component {
  static propTypes = {
    label: PropTypes.string,
  };
  static defaultProps = {
    label: 'default',
  };

  /**
 * Отрисовка компонента
 * @return {ReactElement} Компонент кнопки
 */
  render () {
    return <button>{this.props.label}</button>;
  }
}
