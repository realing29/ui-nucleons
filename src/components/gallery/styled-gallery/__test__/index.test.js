import React from 'react';
import { mount, shallow } from 'enzyme/build';
import StyledGallery, { getControlProps } from '../index';
import BaseGallery from '../../base-gallery/';
import Button from '../../../button';
import Icon from '../../../icon';
import arrowRight from '../../../icons/arrow-right.svg';

describe('<StyledGallery />', () => {
  const items = ['test 1', 'test 2', 'test 3', 'test 4', 'test 5'];
  let gallery;

  it('should render without props', () => {
    gallery = mount(<StyledGallery />);
    expect(gallery.find(BaseGallery)).toHaveLength(1);
    expect(gallery.find(Button)).toHaveLength(0);
    expect(gallery).toMatchSnapshot();
  });
  it('should render with custom props', () => {
    gallery = mount(
      <StyledGallery
        items={items}
        direction='vertical'
        slideStepSize={78}
        slideBy={2}
        galleryContainerProps={{
          className: 'test-gallery',
        }}
        needShowControls='always'
      />
    );
    const galleryContainer = gallery.find('.vertical-gallery-container');
    expect(galleryContainer).toHaveLength(1);
    expect(galleryContainer.hasClass('test-gallery')).toBeTruthy();
    expect(gallery.find(BaseGallery)).toHaveLength(1);
    expect(gallery.find(Button)).toHaveLength(2);
    expect(gallery).toMatchSnapshot();
  });
  it('should render without errors if wrong props value', () => {
    expect(() => shallow(
      <StyledGallery
        items={items}
        galleryContainerProps={null}
        itemsWrapperProps={null}
      />
    )).not.toThrow();
  });
});

describe('getControlProps', () => {
  let controlProps;
  it('default', () => {
    controlProps = getControlProps();
    expect(mount(<Button {...controlProps} />)).toMatchSnapshot();
  });
  it('with params', () => {
    controlProps = getControlProps('backward', true, { backward: [arrowRight, 'test-right-class'] });
    expect(controlProps.children).toEqual(<Icon icon={arrowRight} size={10} />);
    expect(mount(<Button {...controlProps} />)).toMatchSnapshot();
  });
});
