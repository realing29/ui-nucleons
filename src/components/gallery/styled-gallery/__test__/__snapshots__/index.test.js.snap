// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledGallery /> should render with custom props 1`] = `
<StyledGallery
  direction="vertical"
  galleryContainerProps={
    Object {
      "className": "test-gallery",
    }
  }
  items={
    Array [
      "test 1",
      "test 2",
      "test 3",
      "test 4",
      "test 5",
    ]
  }
  needShowControls="always"
  slideBy={2}
  slideStepSize={78}
>
  <div
    className="vertical-gallery-container test-gallery"
  >
    <withGlobalListeners(BaseGallery)
      controlContainer="button"
      direction="vertical"
      getControlProps={[Function]}
      items={
        Array [
          "test 1",
          "test 2",
          "test 3",
          "test 4",
          "test 5",
        ]
      }
      needListenResize={true}
      needShowControls="always"
      slideBy={2}
      slideStepSize={78}
    >
      <BaseGallery
        addGlobalListener={[Function]}
        controlContainer="button"
        direction="vertical"
        getControlProps={[Function]}
        items={
          Array [
            "test 1",
            "test 2",
            "test 3",
            "test 4",
            "test 5",
          ]
        }
        needListenResize={true}
        needShowControls="always"
        slideBy={2}
        slideStepSize={78}
      >
        <button
          aria-label="Промотать назад"
          className="button disabled arrow-up"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <UpSVG>
            <svg
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                clipRule="evenodd"
                d="M12.753 9.659a1 1 0 01-1.412.094L8 6.829 4.659 9.753a1 1 0 01-1.317-1.506l4-3.5a1 1 0 011.317 0l4 3.5a1 1 0 01.094 1.412z"
                fillRule="evenodd"
              />
            </svg>
          </UpSVG>
        </button>
        <div
          className="items-wrapper"
        >
          <div
            className="items-column-container"
            style={
              Object {
                "transform": "translateY(0px)",
                "transition": "all .3s ease 0s",
              }
            }
          >
            <div
              key="0"
            />
            <div
              key="1"
            />
            <div
              key="2"
            />
            <div
              key="3"
            />
            <div
              key="4"
            />
          </div>
        </div>
        <button
          aria-label="Промотать вперед"
          className="button disabled arrow-down"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <DownSVG>
            <svg
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                clipRule="evenodd"
                d="M3.247 6.342a1 1 0 011.412-.095L8 9.171l3.341-2.924a1 1 0 011.318 1.506l-4 3.5a1 1 0 01-1.317 0l-4-3.5a1 1 0 01-.095-1.411z"
                fillRule="evenodd"
              />
            </svg>
          </DownSVG>
        </button>
      </BaseGallery>
    </withGlobalListeners(BaseGallery)>
  </div>
</StyledGallery>
`;

exports[`<StyledGallery /> should render without props 1`] = `
<StyledGallery>
  <div
    className="gallery-container"
  >
    <withGlobalListeners(BaseGallery)
      controlContainer="button"
      direction="horizontal"
      getControlProps={[Function]}
      needListenResize={true}
    >
      <BaseGallery
        addGlobalListener={[Function]}
        controlContainer="button"
        direction="horizontal"
        getControlProps={[Function]}
        needListenResize={true}
      >
        <div
          className="items-wrapper"
        >
          <div
            className="items-container"
            style={
              Object {
                "transform": "translateX(0px)",
                "transition": "all .3s ease 0s",
              }
            }
          />
        </div>
      </BaseGallery>
    </withGlobalListeners(BaseGallery)>
  </div>
</StyledGallery>
`;

exports[`getControlProps default 1`] = `
<ForwardRef(Button)
  aria-label="Промотать вперед"
  className="button"
  disabled={false}
  type="button"
>
  <button
    aria-label="Промотать вперед"
    className="button button-base button-primary button-medium text-button-medium"
    disabled={false}
    type="button"
  >
    <UpSVG>
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          clipRule="evenodd"
          d="M12.753 9.659a1 1 0 01-1.412.094L8 6.829 4.659 9.753a1 1 0 01-1.317-1.506l4-3.5a1 1 0 011.317 0l4 3.5a1 1 0 01.094 1.412z"
          fillRule="evenodd"
        />
      </svg>
    </UpSVG>
  </button>
</ForwardRef(Button)>
`;

exports[`getControlProps with params 1`] = `
<ForwardRef(Button)
  aria-label="Промотать назад"
  className="button disabled test-right-class"
  disabled={true}
  type="button"
>
  <button
    aria-label="Промотать назад"
    className="button disabled test-right-class button-base button-primary button-medium text-button-medium"
    disabled={true}
    type="button"
  >
    <RightSVG>
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          clipRule="evenodd"
          d="M6.341 12.753a1 1 0 01-.094-1.412L9.171 8 6.247 4.659a1 1 0 011.506-1.317l3.5 4a1 1 0 010 1.317l-3.5 4a1 1 0 01-1.412.094z"
          fillRule="evenodd"
        />
      </svg>
    </RightSVG>
  </button>
</ForwardRef(Button)>
`;
